pkg/odo/cli/url/create.go

```diff
@@ -60,17 +64,51 @@ func (o *URLCreateOptions) Complete(name string, cmd *cobra.Command, args []stri
 	} else {
 		o.Context = genericclioptions.NewContext(cmd)
 	}
-	if experimental.IsExperimentalModeEnabled() {
-		if len(args) == 0 {
-			o.urlName = url.GetURLName(o.Component(), o.urlPort)
-		} else {
-			o.urlName = args[0]
-		}
+	if experimental.IsExperimentalModeEnabled() && util.CheckPathExists(o.DevfilePath) {
 		err = o.InitEnvInfoFromContext()
 		if err != nil {
 			return err
 		}
 
+		devObj, err := devfile.Parse(o.DevfilePath)
+		if err != nil {
+			return fmt.Errorf("fail to parse the devfile %s, with error: %s", o.DevfilePath, err)
+		}
+		containers, err := adapterutils.GetContainers(devObj)
+		if err != nil {
+			return err
+		}
+		if len(containers) == 0 {
+			return fmt.Errorf("No valid components found in the devfile")
+		}
+		compWithEndpoint := 0
+		var postList []string
+		for _, c := range containers {
+			if len(c.Ports) != 0 {
+				compWithEndpoint++
+				for _, port := range c.Ports {
+					postList = append(postList, strconv.FormatInt(int64(port.ContainerPort), 10))
+				}
+			}
+			if compWithEndpoint > 1 {
+				return fmt.Errorf("Devfile should only have 1 component contains endpoint")
+			}
+		}
+		if compWithEndpoint == 0 {
+			return fmt.Errorf("No valid component with endpoint found in the devfile")
+		}
+		componentName, _ := component.GetComponentName()
```

Comment:
please don't ignore the err here

Reply:
was a workaround. Now getting the name from envinfo.